### ISSUE TITLE
Feature to specify count of entires for DelimitedBy ()

### ISSUE THREAD

**Issue by 3da** [sprache/Sprache#117]
It should be like:
Parser<IEnumerable<T>> Many<T>(this Parser<T> parser, int maxCount)

And also simmilar new method for DelimitedBy with argument taking maxCount

**Comment by IanWold** [comment]
I think the `Repeat` parser (https://github.com/sprache/Sprache/blob/develop/src/Sprache/Parse.Sequence.cs#L75) is what you're looking for. For your `Many` example, you could just use `Repeat` like such:

```c#
myParser.Repeat(0, maxCount)
```

And you could define your own `DelimitedBy` method using the same concept:

```c#
public static Parser<IEnumerable<T>> DelimitedBy<T, U>(this Parser<T> parser, Parser<U> delimiter, int minimumCount, int maximumCount)
{
    if (parser == null) throw new ArgumentNullException(nameof(parser));
    if (delimiter == null) throw new ArgumentNullException(nameof(delimiter));

    return from head in parser.Once()
           from tail in
               (from separator in delimiter
                from item in parser
                select item).Repeat(minimumCount, maximumCount - 1)
           select head.Concat(tail);
}
```

If others thought it valuable, I would think adding this `DelimitedBy` overload would be acceptable, but I wouldn't support adding an overload for `Many` to achieve this on account that it would just be an alias for `Repeat`.

**Comment by 3da** [comment]
Ok, Repeat is that's what I need for Many(count)

But It would be great for overloaded Delimited with count parameters.

**Comment by 3da** [comment]
I made pull request for this. Somebody will approve or reject?

**Comment by nblumhardt** [comment]
Thanks for the PR; the wheels are turning a bit slowly around here at the moment, but we'll try to get some eyes on it ASAP :-) Cheers!

**Comment by nblumhardt** [comment]
Closed in #120
